src/main/java/org/folio/inventory/resources/TenantItems.java

```diff
@@ -0,0 +1,142 @@
+ */
+public class TenantItems {
+
+  private static final Logger LOG = LogManager.getLogger(MethodHandles.lookup().lookupClass());
+
+  private static final String TENANT_ITEMS_PATH = "/inventory/tenant-items";
+  public static final String ITEMS_FIELD = "items";
+  public static final String TOTAL_RECORDS_FIELD = "totalRecords";
+  public static final String TENANT_ID_FIELD = "tenantId";
+
+  private final HttpClient client;
+
+  public TenantItems(HttpClient client) {
+    this.client = client;
+  }
+
+  public void register(Router router) {
+    router.post(TENANT_ITEMS_PATH + "*").handler(BodyHandler.create());
+    router.post(TENANT_ITEMS_PATH).handler(this::getItemsFromTenants);
+  }
+
+  /**
+   *  This API is meant to be used by UI to fetch different items from several
+   *  tenants together within one call
+   *
+   */
+  private void getItemsFromTenants(RoutingContext routingContext) {
+    var getItemsFutures = routingContext.body().asPojo(TenantItemPairCollection.class)
+      .getTenantItemPairs().stream()
+      .collect(groupingBy(TenantItemPair::getTenantId, mapping(TenantItemPair::getItemId, toList())))
+      .entrySet().stream()
+      .map(tenantToItems -> getItemsWithTenantId(tenantToItems.getKey(), tenantToItems.getValue(), routingContext))
+      .toList();
+
+    CompletableFuture.allOf(getItemsFutures.toArray(new CompletableFuture[0]))
+      .thenApply(v -> getItemsFutures.stream()
+        .map(CompletableFuture::join)
+        .flatMap(List::stream)
+        .toList())
+      .thenApply(this::constructResponse)
+      .thenAccept(jsonObject -> JsonResponse.success(routingContext.response(), jsonObject));
+  }
+
+  private CompletableFuture<List<JsonObject>> getItemsWithTenantId(String tenantId, List<String> itemIds, RoutingContext routingContext) {
+    LOG.info("getItemsWithTenantId:: Fetching items - {} from tenant - {}", itemIds, tenantId);
+    var context = new WebContext(routingContext);
+    CollectionResourceClient itemsStorageClient;
+    try {
+      OkapiHttpClient okapiClient = createHttpClient(tenantId, context, routingContext);
+      itemsStorageClient = createItemsStorageClient(okapiClient, context);
+    }
+    catch (MalformedURLException e) {
+      invalidOkapiUrlResponse(routingContext, context);
+      return CompletableFuture.completedFuture(List.of());
+    }
+
+    var getByIdsQuery = multipleRecordsCqlQuery(itemIds);
+    var itemsFetched = new CompletableFuture<Response>();
+    itemsStorageClient.getAll(getByIdsQuery, itemsFetched::complete);
+
+    return itemsFetched.thenApplyAsync(response ->
+      getItemsWithTenantId(tenantId, response));
+  }
+
+  private List<JsonObject> getItemsWithTenantId(String tenantId, Response response) {
+    if (response.getStatusCode() != HttpStatus.SC_OK || !response.hasBody()) {
+      return List.of();
+    }
+    return JsonArrayHelper.toList(response.getJson(), ITEMS_FIELD).stream()
+      .map(item -> item.put(TENANT_ID_FIELD, tenantId))
```

Comment:
could you say why "tenantId" field is set to the item json? Looks like it does not comply with the response definition because the item schema does not contain "tenantId" field.

Reply:
In screen from testing I see that response object has tenantId populated:
![image](https://github.com/user-attachments/assets/426d2694-d87c-4f75-83e5-804544152274)
Maybe Item allows to add properties dynamically

Reply:
the field "tenantId" successfully has been added to response because in the code we set it to JsonObject representing item and that allows to add any property

Reply:
do you think we can improve this in some way, for example by adding tenantId to Item schema explicitly?

Reply:
I suppose yes, or maybe to add a new response schema for this endpoint that would refer to the item schema